Ne pas oublier **d'exécuter la cellule ci-dessous** pour importer Numpy!

In [3]:
import numpy as np

# I.1. Solution de l'exercice 1

- Créez **deux scalaires** qui ont respectivement pour valeur 5 et 6, et additionnez-les.

- Afficher le **résultat** dans une fonction **print()**.

### <center><u>I.1.1. Solution 1 de l'exercice 1</u></center>

In [4]:
scalaire_1 = np.array(5)
scalaire_2 = np.array(6)
resultat = scalaire_1 + scalaire_2

print(resultat)

11


### <center><u>I.1.2. Solution 2 de l'exercice 1</u></center>

In [6]:
resultat = np.array(5) + np.array(6)

print(resultat)

11


### <center><u>I.1.3. Solution 3 de l'exercice 1</u></center>

Une solution **plus concise**, qui tient en une ligne, mais qui est **moins lisible**.

In [7]:
print(np.array(5) + np.array(6))

11


# I.2. Solution de l'exercice 2

- Créez **trois vecteurs** qui ont pour valeur : [1,2,3], [3,2,1] et [0,1,-1], et additionnez-les.

- Afficher le **résultat** dans une fonction **print()**.

- Afficher en-dessous **la dimension** de ce vecteur.

### <center><u>I.2.1. Solution 1 de l'exercice 2</u></center>

**Solution verbeuse mais explicite**, qui nous permet d'avoir **tous les vecteurs et le résultat en mémoire**.

C'est la solution que je préfère.

In [11]:
vecteur1 = np.array([1,2,3])
vecteur2 = np.array([3,2,1])
vecteur3 = np.array([0,1,-1])

resultat = vecteur1 + vecteur2 + vecteur3
print(resultat)
print(resultat.shape)

[4 5 3]
(3,)


### <center><u>I.2.2. Solution 2 de l'exercice 2</u></center>

Ici, on instancie tous nos vecteurs sur une seule ligne.

Comme on n'a pas stocké le résultat dans une variable, il faut **réeffectuer l'addition** pour afficher la **dimension du vecteur**. Remarquez qu'on peut, grâce aux **parenthèses**, utiliser l'attribut **shape** sur l'addition elle-même.

**Cette solution n'est pas très jolie**, ni pythonique, mais je voulais que vous la voyez pour que vous puissiez entrevoir le champs des possibles avec Python et Numpy.

In [12]:
vecteur1, vecteur2, vecteur3 = np.array([1,2,3]), np.array([3,2,1]), np.array([0,1,-1])

print(vecteur1 + vecteur2 + vecteur3)
print((vecteur1 + vecteur2 + vecteur3).shape)

[4 5 3]
(3,)


### <center><u>I.2.3. Solution 3 de l'exercice 2</u></center>

Cette solution est intéressante car **elle fait un compromis entre une solution très verbeuse et une solution trop courte**. Cependant, le fait de ne pas garder en mémoire les vecteurs individuellement peut poser problème si on veut les réutiliser plus tard.

In [13]:
resultat = np.array([1,2,3]) + np.array([3,2,1]) + np.array([0,1,-1])

print(resultat)
print(resultat.shape)

[4 5 3]
(3,)


# I.3. Solution de l'exercice 3

- Créez un **scalaire** qui a pour valeur : 0
- Créez un **vecteur** qui a pour valeur : [125, 2185, 321, 665, 17, -8]
- **Multipliez le scalaire et le vecteur** (Pour cela, il faut utiliser l'opérateur *)
- Affichez le résultat avec la fonction **print()**
- **Que constatez-vous?**

In [15]:
scalaire = np.array(0)
vecteur = np.array([125, 2185, 321, 665, 17, -8])
resultat = scalaire * vecteur

print(resultat)

[0 0 0 0 0 0]


Le résultat est un **vecteur qui a la même dimension que le vecteur contenu dans la variable vecteur**. Tous ses éléments sont à 0, car tous ses éléments ont été multipliés par le scalaire (qui a une valeur de 0).

# I.4. Solution de l'exercice 4

Une matrice de dimension 2x3, signifie qu'elle possède 2 lignes et 3 colonnes.

In [17]:
m = np.array([[1,2,3], [4,5,6]])
print(m.shape)

(2, 3)


L'objet np.array doit contenir 2 vecteurs qui contiennent chacuns trois valeurs

# I.5. Solution de l'exercice 5

Une matrice de dimension 4x2 signifie qu'elle possède 4 lignes et 2 colonnes.

In [18]:
m1 = np.array([[1,2], [3,4], [5,6], [7,8]])
m2 = np.array([[8,7], [6,5], [4,3], [2,1]])
resultat = m1 + m2
print(resultat)

[[9 9]
 [9 9]
 [9 9]
 [9 9]]


On obtient **une matrice de dimension 4x2**. Chaque élément à la position (m,n) de la première matrice a été additionné avec l'élément à la position (m,n) de la seconde matrice.

On peut vérifier avec l'attribut **shape** que c'est bien une matrice de dimension 4x2.

In [19]:
m1 = np.array([[1,2], [3,4], [5,6], [7,8]])
m2 = np.array([[8,7], [6,5], [4,3], [2,1]])
resultat = m1 + m2
print(resultat.shape)

(4, 2)


# I.6. Solution de l'exercice 6

### <center><u>I.6.1. Solution 1 de l'exercice 6</u></center>

In [20]:
#2. On définit les vecteurs
v1 = np.array([4,7])
v2 = np.array([3,6])

#3. On définit les scalaires
a1 = np.array(2)
a2 = np.array(3)
a3 = np.array(1.5)

#4. On multiplie le vecteur v1 par a1
v1 = v1 * a1

#5. On multiplie le vecteur v2 par a2
v2 = v2 * a2

#6. Addition des deux vecteurs
v3 = v1 + v2

#7. On multiplie le vecteur v3 par a3
resultat = v3 * a3

#8. On affiche le résultat
print(resultat)

[25.5 48. ]


On obtient le **vecteur de dimension 2** : [25.5, 48]

### <center><u>I.6.2. Solution 2 de l'exercice 6</u></center>

Dans ce code, on déclare les vecteurs et les scalaires sur une seule ligne.

On utilise l'opérateur ***=** pour **multiplier** chaque vecteur par le scalaire correspondant.

**Rappel :** v3 = v3 * a3 est équivalent à v3 *= a3

In [23]:
#2. On définit les vecteurs
v1, v2 = np.array([4,7]), np.array([3,6])

#3. On définit les scalaires
a1, a2, a3 = np.array(2), np.array(3), np.array(1.5)

#4. On multiplie le vecteur v1 par a1
v1 *= a1

#5. On multiplie le vecteur v2 par a2
v2 *= a2

#6. Addition des deux vecteurs
v3 = v1 + v2

#7. On multiplie le vecteur v3 par a3
resultat = v3 * a3

#8. On affiche le résultat
print(resultat)

[25.5 48. ]


**Le même code sans les commentaires.**

In [24]:
v1, v2 = np.array([4,7]), np.array([3,6])
a1, a2, a3 = np.array(2), np.array(3), np.array(1.5)
v1 *= a1
v2 *= a2
v3 = v1 + v2
resultat = v3 * a3
print(resultat)

[25.5 48. ]
